src/winmenu.c

```diff
@@ -71,14 +71,19 @@ Menu *CreateWindowMenu(ClientNode *np)
       AddWindowMenuItem(menu, _("Minimize"), MA_MINIMIZE, np, 0);
    }
 
-   if(!(np->state.status & STAT_FULLSCREEN)) {
+   if(!(np->state.status & STAT_FULLSCREEN) && 
+      !(np->state.status & STAT_MINIMIZED)  && 
+       (np->state.border & BORDER_MIN))
+   {
```

Comment:
I think the goal here is to hide the shade/unshade options if either the window is minimized or if it does not support minimization, correct?  That seems reasonable to me, but this will also prevent maximization in that case, which I don’t think is desirable.  There should probably be another `if` statement surrounding just the shade portion.